packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,868 @@
+      txHash,
+      eventTopics: this.eventTopics.splitUpdated,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      return {
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async predictDeterministicAddress(
+    createSplitArgs: CreateSplitV2Config,
+  ): Promise<{
+    splitAddress: Address
+  }> {
+    if (!createSplitArgs.controller) createSplitArgs.controller = zeroAddress
+    if (!createSplitArgs.creator) createSplitArgs.creator = zeroAddress
+
+    const { addresses, allocations } = getAddressAndAllocationFromRecipients(
+      createSplitArgs.recipients,
+    )
+    const totalAllocation = PERCENTAGE_SCALE
+
+    validateAddress(createSplitArgs.controller)
+    validateAddress(createSplitArgs.creator)
+    addresses.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+
+    const factory = this._getSplitV2FactoryContract(
+      createSplitArgs.splitType ?? SplitV2Type.Pull,
+    )
+
+    let splitAddress
+    if (createSplitArgs.salt) {
+      splitAddress = await factory.read.predictDeterministicAddress([
+        {
+          recipients: addresses,
+          allocations,
+          totalAllocation: totalAllocation,
+          distributionIncentive: getNumberFromPercent(
+            createSplitArgs.distributorFeePercent,
+          ),
+        },
+        createSplitArgs.controller,
+        createSplitArgs.salt,
+      ])
+    } else {
+      splitAddress = await factory.read.predictDeterministicAddress([
+        {
+          recipients: addresses,
+          allocations,
+          totalAllocation: totalAllocation,
+          distributionIncentive: getNumberFromPercent(
+            createSplitArgs.distributorFeePercent,
+          ),
+        },
+        createSplitArgs.controller,
+      ])
+    }
+
+    return {
+      splitAddress,
+    }
+  }
+
+  async isDeployed(createSplitArgs: CreateSplitV2Config): Promise<{
+    splitAddress: Address
+    deployed: boolean
+  }> {
+    if (!createSplitArgs.controller) createSplitArgs.controller = zeroAddress
+    if (!createSplitArgs.creator) createSplitArgs.creator = zeroAddress
+
+    const { addresses, allocations } = getAddressAndAllocationFromRecipients(
+      createSplitArgs.recipients,
+    )
+
+    validateAddress(createSplitArgs.controller)
+    addresses.map((recipient) => validateAddress(recipient))
+    this._requirePublicClient()
+
+    const factory = this._getSplitV2FactoryContract(
+      createSplitArgs.splitType ?? SplitV2Type.Pull,
+    )
+
+    const totalAllocation = PERCENTAGE_SCALE
+
+    if (!createSplitArgs.salt) throw new Error('Salt required')
+    const [splitAddress, deployed] = await factory.read.isDeployed([
+      {
+        recipients: addresses,
+        allocations,
+        totalAllocation: totalAllocation,
+        distributionIncentive: getNumberFromPercent(
+          createSplitArgs.distributorFeePercent,
+        ),
+      },
```

Comment:
nit: might be helpful to have a helper function for generating these args from the input? think `createSplit`, `predictDeterministicAddress` and `isDeployed` all are pretty similar? Maybe they are different enough though that a helper function would cause more work/pain. Just wanted to throw out there.